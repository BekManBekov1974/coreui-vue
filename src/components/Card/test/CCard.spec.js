import { mount } from '@vue/test-utils'
import Component from "../CCard";
const ComponentName = 'CCard'

const wrapper = mount(Component, {
  context: {
    props: {
      align: 'left',
      variant: 'success',
      textVariant: 'white',
      borderVariant: 'solid',
      header: 'header',
      footer: 'footer',
      body: 'body'
    }
  }
})
const slotWrapper = mount(Component, {
  context: {
    props: {
      align: 'left',
      variant: 'success',
      textVariant: 'white',
      borderVariant: 'solid',
      header: 'header',
      footer: 'footer',
      body: 'body'
    }
  },
  slots: {
    default: 'this should overwrite body<br>'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('renders correctly with slots', () => {
    expect(slotWrapper.element).toMatchSnapshot()
  })
});
