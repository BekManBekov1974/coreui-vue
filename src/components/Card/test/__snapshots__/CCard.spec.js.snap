// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CCard renders correctly 1`] = `
<div
  class="c-card c-text-left c-bg-success c-border-solid c-text-white"
>
  <!---->
  <!---->
</div>
`;

exports[`CCard renders correctly with slots 1`] = `
<div
  class="c-card c-text-left c-bg-success c-border-solid c-text-white"
>
  <!---->
  this should overwrite body
  <br />
  <!---->
</div>
`;
